### PR TITLE
Added FileOverview to UpdateOverviewResponse 

### PR DESCRIPTION
**Pull request by oliveromahony**
### Proposed changes

Added FileOverview to respond to the Upload Config. This gives a list of files uploaded successfully in the response back to the management plane, so that the Management Plane can reconcile the data has been uploaded

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [x] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

**Review by yluf5 — COMMENTED** [review]
Tested with N1 changes.  It is working as expected.

**Review by craigell — APPROVED** [review]

File: internal/file/file_manager_service.go [hunk 1]
```diff
@@ -87,18 +89,22 @@ func (fms *FileManagerService) UpdateOverview(
 	ctx context.Context,
 	instanceID string,
 	filesToUpdate []*mpi.File,
+	iteration int,
 ) error {
+	const maxAttempts = 5
```
Comment (dhurley) [inline internal/file/file_manager_service.go:94]: Should the const be declared outside of the function?

File: internal/file/file_manager_service.go [hunk 2]
```diff
@@ -147,9 +153,57 @@ func (fms *FileManagerService) UpdateOverview(
...
+
+	for _, file := range diffFiles {
+		updateErr := fms.UpdateFile(ctx, instanceID, file)
+		if updateErr != nil {
+			return updateErr
+		}
+	}
+
+	iteration++
+	slog.Debug("iteration value", "iteration", iteration)
```
Comment (dhurley) [inline internal/file/file_manager_service.go:202]:
```suggestion
	slog.Debug("Updating file overview", "attempt_number", iteration)
```

File: internal/logger/logger.go [hunk 3]
```diff
@@ -132,7 +132,7 @@ func GetCorrelationIDAttr(ctx context.Context) slog.Attr {
 	value, ok := ctx.Value(CorrelationIDContextKey).(slog.Attr)
 	if !ok {
 		slog.Debug("Correlation ID not found in context")
```
Comment (dhurley) [inline internal/logger/logger.go:134]:
```suggestion
        correlationID := GenerateCorrelationID()
		slog.Debug("Correlation ID not found in context, generating new correlation ID", "correlation_id", correlationID)
```

**Review by dhurley — APPROVED** [review]

**Review by sean-breen — APPROVED** [review]

**Pull request merged**